### PR TITLE
Remove Scopes and Add Viewer Role

### LINKED ISSUE #2776: Automated Migration Plan: Scopes -> Roles
### Is your feature request related to a specific problem?

Need a better migration plan for Access Controls.  Users will abruptly lose their scopes UI, while they still have their existing scopes. We're moving to Role Based Access Controls and need a good plan to transition users. 

### Describe the solution you'd like

Decision from 3/9/22 meeting with @NevilleS , @seanpreston , @mfbrown , @pattisdr , @rsilvery

Decided we would add a data migration to automatically move all users to a `viewer` role and add a warning to contact their Root user to have their role assigned.

- Existing scopes should be removed from the user permissions table (leave column in place for now)
- Existing scopes should be cleared from the associated client table (only those clients associated with users). 
- Users will need to log out and log back in. 

A followup in a future sprint will deprecate users being assigned scopes directly and get rid of the userpermission.scopes column. https://github.com/ethyca/fides/issues/2696

Some customers will be contacted directly 

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by pattisdr**
Closes #2776 
❗ Locked to the same release as https://github.com/ethyca/fides/pull/2739

### Code Changes

* [ ] Removes CONFIG_UPDATE from `contributor` scopes - https://github.com/ethyca/fides/issues/2791#issuecomment-1465302052
* [ ] Adds a data migration to give all users a `viewer` role, removes their scopes from their permissions and client table
* [ ] Tacks on automatically generated `messagingconfig_service_type_key` unique index, leftovers from 179f2bb623ae_update_table_for_twilio.py. From Adam: i'm assuming that index got deleted in my migration along with dropping the column it refers to, i.e. with op.drop_column("messagingconfig", "service_type_old")  - since earlier in the migration i had: `op.alter_column(
        "messagingconfig", "service_type", new_column_name="service_type_old"
    )`
and so now the new migration is adding it back in to keep things in sync
* [ ] Updatest src/fides/data/test_env/fides.test_env.toml since we now develop with the more strict `prod` setting - this extends this to when we run fides test env

### Steps to Confirm

* [ ] _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

This is an automated data migration to transition to the new "roles".

Sets all user scopes to an empty list and gives all users a `viewer` role.  Someone will need to login as a root user and assign  roles.  Intentionally nothing preventing users from adding scopes to their users via the API yet.

A followup in a future sprint will get rid of `fidesuserpermission.scopes` entirely https://github.com/ethyca/fides/issues/2696

### CODE REVIEW

**Review by allisonking — APPROVED** [review]
the scopes/roles migration lgtm! don't know much about the other stuff though :)

File: src/fides/api/ctl/migrations/versions/50180bbbb959_automigrate_viewer_role.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from sqlalchemy import text
+
+revision = "50180bbbb959"
+down_revision = "68a518a3c050"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.drop_constraint(
```
Comment (adamsachs) [inline src/fides/api/ctl/migrations/versions/50180bbbb959_automigrate_viewer_role.py:21]: 🙏 thank you for including this!

File: src/fides/lib/oauth/roles.py [hunk 2]
```diff
@@ -114,6 +115,7 @@ class RoleRegistryEnum(Enum):
     MESSAGING_CREATE_OR_UPDATE,
     MESSAGING_DELETE,
     PRIVACY_REQUEST_NOTIFICATIONS_CREATE_OR_UPDATE,
+    CONFIG_UPDATE,
```
Comment (adamsachs) [inline src/fides/lib/oauth/roles.py:118]: 🙏 and thank you for including this too!

**Review by adamsachs — APPROVED** [review]
this looks good, thanks a lot for the cleanup/follow-up work here! 🧹

**Comment by pattisdr** [comment]
Updated with main for CHANGELOG conflict

**Pull request merged**